config/config.go

```diff
@@ -665,6 +679,18 @@ type MetadataConfig struct {
 	SendInterval model.Duration `yaml:"send_interval"`
 }
 
+// SigV4Config is the configuration for signing remote write requests with
+// AWS's SigV4 verification process.
+type SigV4Config struct {
+	// Enabled uses SigV4 for signing remote write requests. Must not be
+	// true at the same time as using one of the standard remote_write
+	// authentication methods.
+	Enabled bool `yaml:"enabled,omitempty"`
```

Comment:
I'm also not thrilled about an `enabled` field, but it seems to have some existing precedence with MetadataConfig's `send: true`. Maybe this should have a different name?

Reply:
I guess you could maybe use  a pointer for SigV4Config and check if nil ? That way `sigv4:` would enable this?

I guess we also need access_key: and secret_key:, like we do in the service discovery.

Reply:
Yeah, I thought about that, though IMO an empty `sigv4:` looks a little awkward. 

👍 to adding more explicit fields to override the default chain, will do.

Reply:
setting a region but not having it work because you did not put enabled is more awkward I think.

Reply:
I see that we can also bypass the endpoint_url in the aws config, which might be handy to e.g. test locally. Is it doable as well? Would it be the same endpoint?

Reply:
I don't think we need to bypass the endpoint URL to test locally. None of this communicates with AWS and the credentials don't need to be real. 

I'm looking for a Go library that validates a signature, but not finding anything just yet. Worst case scenario we could [validate it ourselves](https://docs.aws.amazon.com/general/latest/gr/sigv4-calculate-signature.html).

Reply:
Yeah I deleted my comment after digging deeper. Thanks!

Reply:
I also lean slightly towards the pointer rather than an enabled field. The MetadataConfig's send flag is something we expect may go away in the future so I would avoid using it as precedence.